### PR TITLE
Add identity id to ophan tracking (and default to signed in user identity ID if email provided) for Paypal payments

### PR DESCRIPTION
**Pull request by jranks123**
Previously, we implemented the functionality whereby we take the email used in a payment, get the identity ID (or create a new one if it doesn't exist) associated with that identity ID and include it in our PostGres tracking. 

This PR adds this to the Ophan tracking, and also allows the client to provide an optional email address of a signed in user to supercede the email used in the payment. This is because if a user is signed in with one email address but uses a different email address for Paypal, we would want to attach the contribution to the email they have chosen to use to sign in with, rather than a possible arbitary email address they use for paypal. 

A possible future feature would be to offer people who have made a payment the chance, after paying, to change the account their contribution is attached to from the one that they paid with.

### CODE REVIEW

**Comment by jbguardian** [comment]
LGTM 👍

File: build.sbt [hunk 1]
```diff
@@ -54,6 +54,7 @@ lazy val root = (project in file("."))
 enablePlugins(SystemdPlugin, PlayService, RoutesCompiler, RiffRaffArtifact, JDebPackaging)
 
 resolvers += Resolver.bintrayRepo("guardian", "ophan")
+resolvers += Resolver.sonatypeRepo("releases")
```
Comment (joelochlann) [inline build.sbt]: why didn't we need this before?

File: src/test/scala/controllers/PaypalControllerSpec.scala [hunk 2]
```diff
@@ -428,7 +428,8 @@ class PaypalControllerSpec extends PlaySpec with Status {
             |    "componentId": "componentId",
             |    "componentType": "AcquisitionsOther",
             |    "source": "GuardianWeb"
-            |  }
+            |  },
+            | "signedInUserEmail": "a@b.com"
```
Comment (joelochlann) [inline src/test/scala/controllers/PaypalControllerSpec.scala]: this indentation is wrong - extra space needed

File: src/main/scala/backend/PaypalBackend.scala [hunk 3]
```diff
@@ -52,13 +52,13 @@ class PaypalBackend(
...
-    ophanService.submitAcquisition(PaypalAcquisition(payment, acquisitionData)).bimap(PaypalBackendError.fromOphanError, _ => ())
+  def submitAcquisitionToOphan(payment: Payment, acquisitionData: AcquisitionData, identityId: Option[Long])(implicit pool: DefaultThreadPool): EitherT[Future, PaypalBackendError, Unit] = {
+    ophanService.submitAcquisition(PaypalAcquisition(payment, acquisitionData, identityId)).bimap(PaypalBackendError.fromOphanError, _ => ())
   }
 
 
-  def trackContribution(payment: Payment, acquisitionData: AcquisitionData)(implicit pool: DefaultThreadPool): EitherT[Future, PaypalBackendError, Unit]  = {
-    getOrCreateIdentityIdFromEmail(payment.getPotentialPayerInfo.getEmail).map(Option(_))
+  def trackContribution(payment: Payment, acquisitionData: AcquisitionData, signedInUserEmail: Option[String])(implicit pool: DefaultThreadPool): EitherT[Future, PaypalBackendError, Unit]  = {
+    getOrCreateIdentityIdFromEmail(signedInUserEmail.getOrElse(payment.getPayer.getPayerInfo.getEmail)).map(Option(_))
```
Comment (joelochlann) [inline src/main/scala/backend/PaypalBackend.scala]: this line is unreadably long/nested - could we please break things out into some intermediate variables?

**Pull request merged**